t/import_csv_file.t

```diff
@@ -0,0 +1,193 @@
+
+use Data::DeepAccess qw(deep_exists deep_get deep_set);
+use File::Basename "dirname";
+use File::Path qw/make_path remove_tree/;
+
+use ProductOpener::Config '$data_root';
+use ProductOpener::Data qw/execute_query get_products_collection/;
+use ProductOpener::Producers qw/load_csv_or_excel_file convert_file/;
+use ProductOpener::Products "retrieve_product";
+use ProductOpener::Store "store";
+use ProductOpener::Test qw/:all/;
+
+
+my $test_id = "import_csv_file";
+my $test_dir = dirname(__FILE__);
+my $inputs_dir = "$test_dir/inputs/$test_id/";
+my $expected_dir = "$test_dir/expected_test_results/$test_id/";
+my $outputs_dir = "$test_dir/outputs/$test_id";
+make_path($outputs_dir);
+
+my $usage = <<TXT
+
+The expected results of the tests are saved in $test_dir/expected_test_results/$test_id
+
+To verify differences and update the expected test results,
+actual test results can be saved by passing --update-expected-results
+
+The directory will be created if it does not already exist.
+
+TXT
+;
+
+
+my $update_expected_results;
+
+GetOptions ("update-expected-results"   => \$update_expected_results)
+  or die("Error in command line arguments.\n\n" . $usage);
+
+# fake image download using input directory instead of distant server
+sub fake_download_image ($) {
+    my $image_url = shift;
+
+    my $fname = (split(m|/|, $image_url))[-1];
+    my $image_path = $inputs_dir . $fname;
+    my $response = qobj(
+        is_success => qmeth { return (-e $fname); },
+        decoded_content => qmeth {
+            open(my $image, "<r", $fname);
+            my $content = <$image>;
+            close $image;
+            return $content;
+        },
+    );
+    return $response;
+}
+
+
+# fields we don't want to check for they vary from test to test
+my @fields_ignore_content = qw(last_modified_t created_t owner_fields sources.0.import_t);
+# fields that are array and need to sort to have predictable results
+my @fields_sort = qw(_keywords);
+
+
+# clean products fields that we can't check because they change over runs
+# we may still add some test on those fields here
+sub clean_products_fields($) {
+    my $array_ref = shift;
+
+    my @missing_fields = ();
+
+    for my $product (@$array_ref) {
+        my $code = $product->{code};
+        my @key;
+        for my $field_ic (@fields_ignore_content) {
+            @key = split(/\./, $field_ic);
+            if (!deep_exists($product, @key)) {
+                push(@missing_fields, ($code, $field_ic));
+            } else {
+                deep_set($product, @key, "--ignore--");
+            }
+        }
+        for my $field_s (@fields_sort) {
+            @key = split(/\./, $field_s);
+            if (!deep_exists($product, @key)) {
+                push(@missing_fields, ($code, $field_s));
+            } else {
+                my @sorted = sort @{deep_get($product, @key)};
+                deep_set($product, @key, \@sorted);
+            }
+        }
+    }
+    if (@missing_fields) {
```

Comment:
I'm not sure this is useful. If we are missing a field that is expected, we will see it in the diff later anyway.

Removing this check would allow the function to be used in more cases, and not having each test to specify which fields to remove.

Reply:
Done.